keras_cv/layers/preprocessing/cut_mix.py

```diff
@@ -0,0 +1,142 @@
+import tensorflow as tf
+import tensorflow.keras as keras
+from tensorflow.keras import layers
+
+
+class CutMix(layers.Layer):
+    """
+    CutMix implements the CutMix data augmentation technique as proposed in https://arxiv.org/abs/1905.04899.
+
+    Args:
+        num_classes: number of classes in the dataset.
+        alpha: alpha parameter for the sample distribution.
+        probability: probability to apply the CutMix augmentation.
+        label_smoothing: coefficient used in label smoothing.
+
+    Sample usage:
+    ```python
+    (x_train, y_train), (x_test, y_test) = keras.datasets.cifar10.load_data()
+    cutmix = CutMix(10)
+    augmented_data, updated_labels = cutmix(x_train, y_train)
+    ```
+    """
+    def __init__(
+        self, num_classes, alpha=0.8, probability=1.0, label_smoothing=0.0, **kwargs
+    ):
+        super(CutMix, self).__init__(*kwargs)
+        self.alpha = alpha
+        self.probability = probability
+        self.num_classes = num_classes
+        self.label_smoothing = label_smoothing
+
+    @staticmethod
+    def _sample_from_beta(alpha, beta, shape):
+        sample_alpha = tf.random.gamma(shape, 1.0, beta=alpha)
+        sample_beta = tf.random.gamma(shape, 1.0, beta=beta)
+        return sample_alpha / (sample_alpha + sample_beta)
+
+    def call(self, images, labels):
+        augment_cond = tf.less(
+            tf.random.uniform(shape=[], minval=0.0, maxval=1.0), self.probability
+        )
+        # pylint: disable=g-long-lambda
+        cutmix_augment = lambda: self._update_labels(*self._cutmix(images, labels))
+        no_augment = lambda: (images, self._smooth_labels(labels))
```

Comment:
Just curious why label smooth is still applied when there is no augmentation.

Reply:
This is done in the model garden implementation, as well as the original paper's implementation.  For some reason it helps regularize the model and it works very well with these augmentation techniques.  It's off by default, so it will at least not surprise users.